db/badger.go

```diff
@@ -10,11 +11,18 @@ import (
 // golang
 type badgerDb struct {
 	badger *badger.DB
+	wMutex *sync.Mutex
```

Comment:
Aren't badger transactions thread safe? So Why do we need to only use one transaction at a time? 

I don't think we need to have a reference to a [mutex](https://github.com/uber-go/guide/blob/master/style.md#zero-value-mutexes-are-valid). Although golang will panic if you were to copy a mutex.

Reply:
> Aren't badger transactions thread safe? So Why do we need to only use one transaction at a time?

that is mostly because Pebble does not have transactions but only `Batch`s. Batchs dont check for conflicts on between different write batches, so that would create race conditions if we were to allow multiple write "Transactions".

So I made it a requirement on our `DB` interface that it allows only a single update txn at a time.

Reply:
> I don't think we need to have a reference to a [mutex](https://github.com/uber-go/guide/blob/master/style.md#zero-value-mutexes-are-valid). Although golang will panic if you were to copy a mutex.

I will take a look, thanks!